src/PrestaShopBundle/Controller/Admin/Improve/International/ZoneController.php

```diff
@@ -0,0 +1,381 @@
+    public function toggleStatusAction(int $zoneId)
+    {
+        try {
+            $this->getCommandBus()->handle(new ToggleZoneStatusCommand($zoneId));
+            $this->addFlash(
+                'success',
+                $this->trans('The status has been successfully updated.', 'Admin.Notifications.Success')
+            );
+        } catch (Exception $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages($e)));
+        }
+
+        return $this->redirectToRoute('admin_zones_index');
+    }
+
+    /**
+     * Deletes zones in bulk action
+     *
+     * @AdminSecurity("is_granted('delete', request.get('_legacy_controller'))", redirectRoute="admin_zones_index")
+     * @DemoRestricted(redirectRoute="admin_zones_index")
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function bulkDeleteAction(Request $request)
+    {
+        $zoneIds = $this->getBulkZonesFromRequest($request);
+
+        try {
+            $this->getCommandBus()->handle(new BulkDeleteZoneCommand($zoneIds));
+
+            $this->addFlash(
+                'success',
+                $this->trans('The selection has been successfully deleted.', 'Admin.Notifications.Success')
+            );
+        } catch (ZoneException $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages($e)));
+        }
+
+        return $this->redirectToRoute('admin_zones_index');
+    }
+
+    /**
+     * Bulk toggles zones status.
+     *
+     * @AdminSecurity(
+     *     "is_granted('update', request.get('_legacy_controller'))",
+     *     redirectRoute="admin_zones_index",
+     * )
+     * @DemoRestricted(redirectRoute="admin_zones_index")
+     *
+     * @param string $status
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function bulkToggleStatus(string $status, Request $request)
+    {
+        $status = $status == 'enable';
+        $zoneIds = $this->getBulkZonesFromRequest($request);
+
+        try {
+            $this->getCommandBus()->handle(new BulkToggleZoneStatusCommand($status, $zoneIds));
+            $this->addFlash(
+                'success',
+                $this->trans('The status has been successfully updated.', 'Admin.Notifications.Success')
+            );
+        } catch (ZoneException $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages($e)));
+        }
+
+        return $this->redirectToRoute('admin_zones_index');
+    }
+
+    private function getErrorMessages(Exception $e)
+    {
+        return [
+            CannotEditZoneException::class => $this->trans(
+                'An error occurred while editing zone.',
+                'Admin.International.Notification'
+            ),
+            InvalidZoneValuesException::class => $this->trans(
+                'Invalid value(s) provided for zone.',
```

Comment:
What invalid values can be provided?

Reply:
Actually there can't be any. Only is empty name but this is handled by another exception. Removed this exception.

Reply:
Alright, thanks a lot!